### PR TITLE
Toxic Comments Plugin

### PR DESCRIPTION
**Pull request by jtnelson**
## What does this PR do?
- Implement toxic comment plugin
- Attach perspective scores to the metadata when posting comments
- Let users reconsider their comment if it contains toxic language.
- Flag comments with toxic language
- Removed `wrapResponse` helper for mutation resolvers and achieve a similar result using the new `errorHandler` schema decorator.
- A comment that is toxic lands in `premod` (will change in a later PR)
- New environment variables (might change in a later PR)
  - `TALK_PERSPECTIVE_API_ENDPOINT` (default https://commentanalyzer.googleapis.com/v1alpha1)
  - `TALK_PERSPECTIVE_API_KEY`
  -  `TALK_TOXICITY_THRESHOLD` (default 0.8)
  - `TALK_PERSPECTIVE_TIMEOUT` (default 300)

## How do I test this PR?

- export TALK_PERSPECTIVE_API_KEY=<api key> # see 1password for test api key
- start Talk
- enter a toxic comment (e.g. one with curse words)
- verify that user is warned 
- finally post the comment, it should not appear in stream (if still toxic)
- Verify toxicity score is attached to comment metadata

### CODE REVIEW

File: plugins/talk-plugin-toxic-comments/server/perspective.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = (perspective) => {
+  console.log("hello world from perpsctive");
```
Comment (okbel) [inline plugins/talk-plugin-toxic-comments/server/perspective.js]: Extra log :)

File: plugins/talk-plugin-toxic-comments/server/router.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const http = require('axios');
+const boom = require('express-boom');
+const bodyParser = require('body-parser');
+var count = 0;
+
+module.exports = (router) => {
+  router.use(boom());
+  router.use(bodyParser.text());
+  router.post('/api/v1/toxicity/comments', (req, res) => {
+    console.log(req.body);
```
Comment (okbel) [inline plugins/talk-plugin-toxic-comments/server/router.js]: Extra log here too

**Review by okbel — DISMISSED** [review]

**Comment by okbel** [comment]
Could you modify the PR description? Thanks!

File: .gitignore [hunk 3]
```diff
@@ -30,5 +30,6 @@ plugins/*
 !plugins/talk-plugin-comment-content
 !plugins/talk-plugin-permalink
 !plugins/talk-plugin-featured-comments
-
+!plugins/talk-plugin-toxic-comments
 **/node_modules/*
+story.html
```
Comment (okbel) [inline .gitignore]: Extra `story.html`

File: plugins/talk-plugin-toxic-comments/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+client
```
Comment (okbel) [inline plugins/talk-plugin-toxic-comments/.gitignore]: Is this needed?

File: plugins/talk-plugin-toxic-comments/server/errors.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const {APIError} = require('../../../errors');
```
Comment (wyattjoh) [inline plugins/talk-plugin-toxic-comments/server/errors.js]: You can load from `require('errors');`

File: plugins/talk-plugin-toxic-comments/server/hooks.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  RootMutation: {
+    createComment: {
+      async pre(_, {input}, _context, _info) {
+
+        // Don't call out to perspective when running tests.
+        if (process.env.NODE_ENV === 'test') {
+          return;
+        }
+
+        const apiKey = require('./apiKey');
```
Comment (wyattjoh) [inline plugins/talk-plugin-toxic-comments/server/hooks.js]: Should not have an inline require, move this to the global scope.

File: plugins/talk-plugin-toxic-comments/server/apiKey.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const apiKey = process.env.TALK_PERSPECTIVE_API_KEY;
+if(!apiKey) {
```
Comment (wyattjoh) [inline plugins/talk-plugin-toxic-comments/server/apiKey.js]: Check out https://github.com/coralproject/talk/blob/master/config.js#L168-L172 to do instead, check if you're in test mode, if you are, don't panic.

File: plugins/talk-plugin-toxic-comments/server/hooks.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+const {TOXICITY_THRESHOLD} = require('./constants');
+const ActionsService = require('../../../services/actions');
+
+module.exports = {
+  RootMutation: {
+    createComment: {
+      async pre(_, {input}, _context, _info) {
+
+        // Don't call out to perspective when running tests.
+        if (process.env.NODE_ENV === 'test') {
```
Comment (wyattjoh) [inline plugins/talk-plugin-toxic-comments/server/hooks.js]: Don't attach the pre hook if we're in test!

File: plugins/talk-plugin-toxic-comments/server/apiKey.js (hunk 7 above)
Comment (alinhle) [reply plugins/talk-plugin-toxic-comments/server/apiKey.js]: Ah.. 👍

**Review by kgardnr — APPROVED** [review]

**Pull request merged**